### PR TITLE
Setup Redux and bug fix with re-renders for charts

### PR DESCRIPTION
**Pull request by Smerly**
Setup Redux and bug fix with re-renders for charts

### CODE REVIEW

File: src/Components/CoinListComponent/EachCoin.js [hunk 1]
```diff
@@ -32,6 +34,12 @@ function EachCoin(props) {
         return daysInAWeek[indexedDate.getDay()]
     })
 
+    const delayChart = () => {
+        setTimeout(() => {
+            return <Line data={data} options={options}/>
+        }, 1000);
+        return <Line data={data} options={options}/>
+    }
```
Comment (mallharpreet) [inline src/Components/CoinListComponent/EachCoin.js]: why did you do this?

Comment (mallharpreet) [reply src/Components/CoinListComponent/EachCoin.js]: find another way please mate

Comment (mallharpreet) [reply src/Components/CoinListComponent/EachCoin.js]: I'm not Harpreet

File: src/redux/portfolioSlice.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+    // currencies interface:
+// {
+//      id: 1,
+//      recordedCurrencyInfo: {currency from api},
+//      amountPurchased: 2,
+//      priceTotalCurrency: $200,
+// }
```
Comment (mallharpreet) [inline src/redux/portfolioSlice.js]: get rid of this

File: src/utils/handleAwait.js [hunk 3]
```diff
@@ -1,5 +1,10 @@
 export const handleAwait = (obj, str) => {
-    return obj ? obj[str] : [1,2,3,4]
+    return typeof obj == 'object' ? obj[str] : [1,2,3,4,5,6,7,8]
+}
+
+export const handleAwaitChart = (obj, str) => {
+    console.log(obj[str])
```
Comment (mallharpreet) [inline src/utils/handleAwait.js]: get rid of this

**Review by mallharpreet — CHANGES_REQUESTED** [review]

File: src/Components/CoinListComponent/EachCoin.js (hunk 1 above)
Comment (Smerly) [reply src/Components/CoinListComponent/EachCoin.js]: Oops, I forgot to get rid of that, that solution to the update error doesn't actually work, I'll get rid of that!

**Review by mallharpreet — APPROVED** [review]

**Pull request merged**